typescript/src/main.ts

```diff
@@ -95,6 +134,21 @@ function appendMetalTableRow(metal: metals.Metal, table: HTMLTableElement) {
   resultCell.id = "result_" + metal.idSuffix;
 }
 
+function updateRow(metal: metals.Metal) {
+  const id = metal.idSuffix;
+  (<HTMLInputElement>document.getElementById("affinity_" + id)).value =
+    metal.affinity.toString();
+  (<HTMLTableCellElement>(
+    document.getElementById("metalation_delta_g_" + id)
+  )).innerText = metal.metalationDeltaG.toFixed(1).toString();
+  (<HTMLInputElement>document.getElementById("bmc_" + id)).value =
+    metal.bufferedMetalConcentration.toString();
+  (<HTMLTableCellElement>(
+    document.getElementById("ia_delta_g_" + id)
+  )).innerText = metal.intracellularAvailableDeltaG.toFixed(1).toString();
+  calculate();
```

Comment:
I don't think we should be calling `calculate` on every row in the reset. Might be better to move this back into the reset function and add it to the toggle button event listener.